src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp

```diff
@@ -451,32 +451,44 @@ HRESULT STDMETHODCALLTYPE CorProfiler::JITCompilationStarted(
   hr = rewriter.Import();
   RETURN_OK_IF_FAILED(hr);
 
-  for (auto& method_replacement : method_replacements) {
-    const auto& wrapper_method_key =
-        method_replacement.wrapper_method.get_method_cache_key();
-    mdMemberRef wrapper_method_ref = mdMemberRefNil;
-
-    if (!module_metadata->TryGetWrapperMemberRef(wrapper_method_key,
-                                                 wrapper_method_ref)) {
-      // no method ref token found for wrapper method, we can't do the
-      // replacement, this should never happen because we always try to
-      // add the method ref in ModuleLoadFinished()
-      // TODO: log this
-      return S_OK;
+  ILInstr* last_instr = nullptr;
+
+  // for each IL instruction
+  for (ILInstr* pInstr = rewriter.GetILList()->m_pNext;
+       pInstr != rewriter.GetILList(); pInstr = pInstr->m_pNext) {
```

Comment:
I swapped the nesting of the loops because having the IL on the outside allows for some short-circuiting optimizations.